### PR TITLE
Log

### PR DESCRIPTION
**Pull request by SharonMosesJangam**
Logger changes after compilation

### CODE REVIEW

**Comment by gaukas** [comment]
logging points looks good. 

I think instead of keeping this branch, you may want to branch from the LATEST master branch and insert those logging points. Doing so effectively prevents merge conflicts.

File: internal/logger/logger.go [hunk 1]
```diff
@@ -1 +1,22 @@
 package logger
```
Comment (gaukas) [inline internal/logger/logger.go:1]: We may need to get rid of this file and use the `logger.go` in the current `master` branch.

File: go.mod [hunk 2]
```diff
@@ -3,3 +3,4 @@ module github.com/gaukas/passthru
 go 1.16
 
 require github.com/refraction-networking/utls v1.1.5
+require go.uber.org/zap v1.19.1
```
Comment (gaukas) [inline go.mod:6]: We don't need it.

File: config/rule.go [hunk 3]
```diff
@@ -44,6 +50,8 @@ func (at *ActionType) MarshalJSON() ([]byte, error) {
 	case ACTION_FORWARD:
 		return []byte("\"FORWARD\""), nil
 	default:
-		return nil, fmt.Errorf("invalid action type: %d", *at)
+		//return nil, fmt.Errorf("invalid action type: %d", *at)
+		return nil, log.Info("invalid action type", zap.String(":", *at))
```
Comment (gaukas) [inline config/rule.go:54]: All returning `fmt.Errorf()` should be kept AS-IS. They are returning an `error` type, not printing to log.

File: config/rule.go [hunk 4]
```diff
@@ -1,6 +1,11 @@
 package config
 
-import "fmt"
+import (
+	"fmt",
+	"go.uber.org/zap",
+	"go.uber.org/zap/zapcore",
```
Comment (gaukas) [inline config/rule.go:6]: don't need to introduce external packages.

**Review by gaukas — CHANGES_REQUESTED** [review]
Changes requested:
- No need to introduce uber packages
- Should keep `fmt.Errorf` AS-IS
- Need to be based on latest master branch instead.

File: internal/logger/logger.go (hunk 1 above)
Comment (SharonMosesJangam) [reply internal/logger/logger.go:1]: sure..I'll do that tonight

File: config/rule.go (hunk 4 above)
Comment (SharonMosesJangam) [reply config/rule.go:6]: I'll remove these changes

**Comment by SharonMosesJangam** [comment]
Updated the PR

**Comment by gaukas** [comment]
Anyway we can get rid of the [.DS_Store](https://github.com/gaukas/passthru/pull/4/files#diff-e2eb93a61ffd7877ea5c751abcb3a618e8e2e9a2073a27f66d4114fe10819f86)?

**Comment by gaukas** [comment]
Don't worry, I will do that

**Comment by SharonMosesJangam** [comment]
I have removed the .DS_Store

**Review by gaukas — APPROVED** [review]

**Comment by SharonMosesJangam** [comment]
Merging Logger-related changes.

**Pull request merged**